bot-components/Bot_info.ml

```diff
@@ -2,19 +2,22 @@ open Base
 
 type t =
   { gitlab_instances: (string, string * string) Hashtbl.t
-  ; github_pat: string
   ; github_install_token: string option
   ; github_name: string
   ; email: string
   ; domain: string
   ; app_id: int }
 
+(* Returns the GitHub installation token. Requires installation token to be set. *)
 let github_token bot_info =
   match bot_info.github_install_token with
   | Some t ->
       t
   | None ->
-      bot_info.github_pat
+      (* TODO: use Result.t later *)
+      failwith
+        "GitHub installation token is required. Please ensure the GitHub App \
+         is installed and an installation token is available."
```

Comment:
Avoid leaving exceptions in finished code except in very specific cases. Here, my suggestion is that instead of handling the case where `github_install_token` is `None`, we just make it a mandatory parameter, just like `github_name`, etc.
OTOH, `github_pat` is the one which we want to make optional, just for the few legitimate uses (that are all specific to the Rocq minimization feature). Since the function `github_token` then becomes trivial, it can just be replaced by direct use of `.github_install_token`.